### PR TITLE
Fixed Imagine Quiz

### PR DESCRIPTION
**Pull request by dm9965**
- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -74,70 +73,81 @@ const App = () => {
...
+              path={`/Lab${lab}/About`}
+              user={state.main.user}
+              labID={lab}
+            />
+
+            <Reading
+              path={`/Lab${lab}/Reading`}
+              user={state.main.user}
+              labID={lab}
+              isImagine={state.main.isImagine}
```
Comment (heathermoses) [inline client/src/App.js]: I would make this `isImagine={isImagine}` since you already defined the constant on line 76

File: client/src/App.js [hunk 2]
```diff
@@ -74,70 +73,81 @@ const App = () => {
...
+            />
+
+            <Quiz
+              path={`/Lab${lab}/Quiz`}
+              labId={lab}
+              user={state.main.user}
+              isFinalQuiz
+              hideCertificate={false}
+              submitData={() => {}}
+              isImagine={state.main.isImagine}
```
Comment (heathermoses) [inline client/src/App.js]: I would make this `isImagine={isImagine}` since you already defined the constant on line 76

**Review by heathermoses — APPROVED** [review]
I tested going through the imagine lab and the regular lab 2 and made sure they both work. Everything looks good to me

File: client/src/components/body/Reading/Reading.js [hunk 3]
```diff
@@ -25,7 +51,49 @@ const Reading = (props) => {
...
+        );
+      }, 1000);
+
+      return () => {
+        clearInterval(interval);
+      };
+    }
+
+    if (isImagine && saveData) {
+      ImagineService.readingSectionPagePosition(userID, pagePosition);
```
Comment (AndreasLc1103) [inline client/src/components/body/Reading/Reading.js:84]: I am a little confused on why we still pass back the userID and user when user contains the userID

File: client/src/components/imagine23/Main.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { navigate, Router } from "@reach/router";
 import { bindActionCreators } from "redux";
 import LandingPage from "./pages/landingPage";
 import MainInstructions from "./pages/mainInstructions";
-import Reading from "./pages/Reading";
+import { default as Reading } from "../body/Reading/Reading";
```
Comment (AndreasLc1103) [inline client/src/components/imagine23/Main.js]: you also could just have it export it as default from the file so you dont have to deal with this destructing in the import

**Review by AndreasLc1103 — CHANGES_REQUESTED** [review]
This needs some minor TLC. if you could explain what userID vs using the User Object would be good. Also fix the import to follow the right es6 standard

**Comment by AndreasLc1103** [comment]
Also make sure you pull in imagine 2024 to fix those issues

File: client/src/components/body/Reading/Reading.js (hunk 3 above)
Comment (dm9965) [reply client/src/components/body/Reading/Reading.js:84]: The userID generated for Imagine is different than the state.main.user created in App.js. userID is getting passed in from the Imagine main.js file when it's created on the UserID page.

File: client/src/components/imagine23/Main.js (hunk 4 above)
Comment (dm9965) [reply client/src/components/imagine23/Main.js]: This has been handled.

**Pull request merged**